### PR TITLE
add update-merkle-root script

### PR DESCRIPTION
**Pull request by MixailE**

### CODE REVIEW

**Comment by MixailE** [comment]
[Add mekleRoot generator script into ketl-attestation-token repo](https://trello.com/c/JpI8ehck/3100-add-mekleroot-generator-script-into-ketl-attestation-token-repo)

File: scripts/update-merkle-root.ts [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+enum VerificationType {
+  email = '0',
+  twitter = '1',
+  balance = '2',
+  token = '3',
+}
+
+function generateHashByRecord(
+  str: string,
+  hashFunc: (message: (number | string | BigNumber)[] | Uint8Array) => string
```
Comment (T-Damer) [inline scripts/update-merkle-root.ts:41]: Let's extract the type

File: scripts/update-merkle-root.ts [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+) {
+  const [verification, content] = str.split(':')
+  switch (verification) {
+    case Verification.token:
+      if (isNaN(parseInt(content, 10)))
```
Comment (T-Damer) [inline scripts/update-merkle-root.ts:46]: Let's extract this to into helper function with proper name

File: scripts/update-merkle-root.ts [hunk 3]
```diff
@@ -0,0 +1,180 @@
...
+  twitter = '1',
+  balance = '2',
+  token = '3',
+}
+
+function generateHashByRecord(
+  str: string,
+  hashFunc: (message: (number | string | BigNumber)[] | Uint8Array) => string
+) {
+  const [verification, content] = str.split(':')
```
Comment (T-Damer) [inline scripts/update-merkle-root.ts:43]: `content` is `address`?

File: scripts/update-merkle-root.ts [hunk 4]
```diff
@@ -0,0 +1,180 @@
...
+      return hashFunc([
+        VerificationType.balance,
+        hexlifyString(content.toLowerCase()),
+        1,
+        hexlifyString(YC_ALUM_NFT_CONTRACT),
+      ])
+    case Verification.BWLNFT:
+      if (!ethereumAddressRegex.test(content.toLowerCase()))
+        throw new Error(`Invalid ethereum address: ${content}!`)
+      return hashFunc([
```
Comment (T-Damer) [inline scripts/update-merkle-root.ts:67]: Let's extract this as well

File: scripts/update-merkle-root.ts [hunk 5]
```diff
@@ -0,0 +1,180 @@
...
+    attestationAddress,
+    deployer
+  )
+
+  const hashFunc = await poseidonHash()
+  const ids = [0, 1, 2, 3]
+
+  for (const id of ids) {
+    const filePath = resolve(cwd(), 'merkleTrees', `${id}.txt`)
+    const liner = new lineByLine(filePath)
```
Comment (T-Damer) [inline scripts/update-merkle-root.ts:123]: `lineByLine` is a class, import and use with UpperCase, please

File: scripts/update-merkle-root.ts [hunk 6]
```diff
@@ -0,0 +1,180 @@
...
+
+    let line
+    const attestationHashes = [] as string[]
+    while ((line = liner.next())) {
```
Comment (T-Damer) [inline scripts/update-merkle-root.ts:127]: Why don't we do `while (const line = liner.next())`?

**Review by T-Damer — APPROVED** [review]

**Review by smyzin — APPROVED** [review]
After resolving comments from Daniil can be merged, imo

File: scripts/update-merkle-root.ts (hunk 3 above)
Comment (MixailE) [reply scripts/update-merkle-root.ts:34]: It could be twitter id, address

File: scripts/update-merkle-root.ts (hunk 6 above)
Comment (MixailE) [reply scripts/update-merkle-root.ts:97]: Nope, it doesn't work, sorry

**Pull request merged**